### ISSUE TITLE
Training stopping early

### ISSUE THREAD

**Issue by MasIgor** [ultralytics/ultralytics#21123]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi all!

i am training a test network with patience=300 and epochs = 1000. 

after 309 epochs it stops with this message: 

```
    Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
   308/1000      11.2G    0.08926    0.09501     0.7753         11        640: 1
                 Class     Images  Instances      Box(P          R      mAP50  m
                   all        771          1          1          1      0.995      0.995

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
   309/1000      12.8G    0.08753    0.09196     0.7731         14        640: 1
                 Class     Images  Instances      Box(P          R      mAP50  m
                   all        771          1          1          1      0.995      0.995
EarlyStopping: Training stopped early as no improvement observed in last 300 epochs. Best results observed at epoch 9, best model saved as best.pt.
To update EarlyStopping(patience=300) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping.
```

but epoch 9 hast a Precision value of only 0.28.. it is only looking at mAP95.. 

```
      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     9/1000      12.8G     0.2714     0.2443     0.8828         11        640: 1
                 Class     Images  Instances      Box(P          R      mAP50  m
                   all        771          1       0.28          1      0.995      0.995
```

why is the epoch 9 seen as teh best if epoch 309 is better?

any help is greatly appreciated..
thanks!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MasIgor, thank you for reaching out and sharing your training results with Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will review your issue and provide further assistance soon.

We recommend checking the [Docs](https://docs.ultralytics.com/) for detailed guidance, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions are already answered there.

Since this appears to be a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you haven't already. This will help us diagnose the issue more efficiently.

If your question relates to custom training, please share additional details such as dataset samples and full training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community for more support:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and learn from others

## Upgrade

First, please upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure the issue isn't already resolved:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for helping improve Ultralytics! 🥇

**Comment by Y-T-G** [comment]
It's based on fitness 

https://github.com/ultralytics/ultralytics/issues/16358#issuecomment-2360276418
